File: wouldproxy.rb
#!/usr/bin/env ruby

require 'net/http'

ARGV.each do |u|
  uri = URI(u)
  h = Net::HTTP.new(uri.host, uri.port)
  if h.proxy?
    puts "would use '#{h.proxy_address}:#{h.proxy_port}' for '#{u}'"
  else
    puts "would not use proxy for '#{u}'"
  end
end

File: wouldproxy.py
#!/usr/bin/env python

import requests
from sys import argv

for u in argv[1:]:
    eproxy = requests.utils.get_environ_proxies(u) 
    if 'http' in eproxy:
        print('would use proxy {proxy} for {url}'.format(url=u, proxy=eproxy['http']))
    else:
        print('would not use proxy for {url}'.format(url=u))
